File: aoj/volume0/n0022/Main.java
import java.util.Scanner;

public class Main {

    public static void main(String[] args) {

        Scanner sc = new Scanner(System.in);

        while (true) {

            int n = sc.nextInt();

            if (n == 0) {
                break;
            }

            long sum = 0;
            long max = Long.MIN_VALUE;

            for (int i = 0; i < n; i++) {

                int a = sc.nextInt();
                sum = Math.max(sum + a, a);
                max = Math.max(max, sum);

            }

            System.out.println(max);

        }

    }

}

File: aoj/volume0/n0022/Main.cpp
#include <iostream>

using namespace std;

int main() {

    while (1) {

        int n;
        cin >> n;

        if (n == 0) {
            break;
        }

        int sum = 0;
        int max;

        for (int i = 0; i < n; i++) {

            int a;
            cin >> a;

            if (i == 0) {
                max = a;
            }

            if (a < (sum + a)) {
                sum += a;
            } else {
                sum = a;
            }

            if (max < sum) {
                max = sum;
            }

        }

        cout << max << endl;

    }

    return 0;

}
